src/main/java/de/tum/in/www1/artemis/web/rest/programming/ProgrammingExerciseParticipationResource.java

```diff
@@ -319,51 +326,60 @@ else if (isTemplateRepository) {
     @EnforceAtLeastStudent
-    public ModelAndView redirectGetParticipationRepositoryFilesForCommitsDetailsView(@PathVariable long exerciseId, @PathVariable long participationId,
-            @PathVariable String commitId, @RequestParam RepositoryType repositoryType) {
-        if (repositoryType == null) {
-            throw new BadRequestAlertException("Invalid repository type", ENTITY_NAME, "invalidRepositoryType");
-        }
-        ProgrammingExerciseParticipation participation;
-        if (repositoryType.equals(RepositoryType.USER)) {
-            participation = programmingExerciseStudentParticipationRepository.findByIdElseThrow(participationId);
-        }
-        else {
-            if (repositoryType.equals(RepositoryType.TEMPLATE)) {
-                participation = programmingExerciseParticipationService.findTemplateParticipationByProgrammingExerciseId(exerciseId);
+    public ResponseEntity<Map<String, String>> getParticipationRepositoryFilesForCommitsDetailsView(@PathVariable long exerciseId,
+            @RequestParam(required = false) Long participationId, @PathVariable String commitId, @RequestParam(required = false) RepositoryType repositoryType)
+            throws GitAPIException, IOException {
+        if (participationId != null) {
+            Participation participation = participationRepository.findByIdElseThrow(participationId);
+            if (!participation.getExercise().getId().equals(exerciseId)) {
+                throw new ConflictException("A git diff report entry can only be retrieved if the participation's exercise id matches with the exercise id", ENTITY_NAME,
+                        "exerciseIdsMismatch");
+            }
+            if (participation instanceof ProgrammingExerciseParticipation programmingExerciseParticipation) {
+                ProgrammingExercise programmingExercise = programmingExerciseRepository.getProgrammingExerciseFromParticipation(programmingExerciseParticipation);
+                participationAuthCheckService.checkCanAccessParticipationElseThrow(participation);
+                // we only forward the repository type for the test repository, as the test repository is the only one that needs to be treated differently
+                return ResponseEntity.ok(repositoryService.getFilesContentAtCommit(programmingExercise, commitId, null, programmingExerciseParticipation));
             }
             else {
-                // if the repository is TESTS we also want to get the solution participation check to see if the user
-                // has access to the participation, as the TESTS repository doesn't have a participation
-                participation = programmingExerciseParticipationService.findSolutionParticipationByProgrammingExerciseId(exerciseId);
+                throw new ConflictException("A git diff report entry can only be retrieved for programming exercise participations", ENTITY_NAME,
+                        "notProgrammingExerciseParticipation");
```

Comment:
Isn't that more or less the code from `ProgrammingExerciseGitDiffResourse.getVcsRepositoryUri`?

Reply:
I struggled a bit with this one - there didn't seem to be a great option for resolving this. If you have a suggestion, please let me know and I'll implement it.

I moved the method to a service and call it from both locations now